src/fetcher/query.ts

```diff
@@ -1,400 +1,549 @@
-// TODO: needs type declaration
-export const idsQuery = (ids: any[]) => {
-  const query = ids.map(
-    (id) => `
-    uuid${id}: uuid(id:${id}) {
-        ... on Entity {
-          alias
-        }
-        ... on Page {
-          alias
-        }
-        ... on TaxonomyTerm {
-          alias
-        }
-      }
-  `
-  )
-  return `{${query}}`
+// Keep this file in sync with the graphQL schema.
+// Maybe automate this one day.
+
+export type Instance = 'de' | 'en' | 'fr' | 'es' | 'ta' | 'hi'
+
+// A license has some more attributes, but we are fine with these
+
+const license = `
+  license {
+    id
+    url
+    title
+  }
+`
+
+export interface License {
+  id: number
+  url: string
+  title: string
 }
 
-// TODO: needs type declaration
-export const idQuery = (id: any) => `
-  {
-    uuid(id:${id}) {
-      ... on Entity {
-        alias
-      }
-      ... on Page {
-        alias
-      }
-      ... on TaxonomyTerm {
-        alias
-      }
+// This is one breadcrumb path. Why is only url optional?
```

Comment:
?

Reply:
![grafik](https://user-images.githubusercontent.com/13507950/87020501-371b8780-c1d4-11ea-80fe-3364678264ab.png)

Label is required, but url is not. This means we can have navigation nodes with label, but without url.

Reply:
Basically: every navigation node has a required `label` and either `url` or `id`. `url` would be set if this is a custom route, e.g. `/auth/login`. `id` would be set if it is an Uuid (and `/<id>` resp. `/<alias>` could be used as an url).

Reply:
I am a bit confused. As far as I understand, NavigationNode is an entry in the breadcrumbs, which has no Dropdown functionality? The secondary navigation, that contained nested elements, is stored separately in navigation.data (and is passed down as JSON)

So I think, in this case `url` can be made required??

Reply:
The `NavigationNode` in breadcrumbs is just another perspective on the on the navigation. So if the currently viewed page is a child of a dropdown, the dropdown element will appear in the breadcrumbs too, e.g.
```
Mathematik
  > Some Dropdown
     > Some Page
```

Reply:
How is this related to the taxonomy? I thought that breadcrumbs are only using paths from the taxonomy (and please don't tell me that the taxonomy is just another perspective of the navigation :smile:)

Reply:
Basically: the Taxonomy is part of the navigation and the breadcrumbs may contain information from both navigation and taxonomy tree. See e.g. for uuid 1855, the breadcrumbs are:
```
Mathematik (Navigation element that points to the math landing page /19767)
> Alle Themen (Navigation element that points to the math subject /5)
> Funktionen (TaxonomyTerm)
> Wichtige Funktionstypen und ihre Eigenschaften (TaxonomyTerm)
> Quadratische Funktionen - Parabeln (TaxonomyTerm)
> Funktionsterm einer quadratischen Funktion (TaxonomyTerm)
```
So basically, we glue together the navigation data to the subject and the taxonomy term path to the entity. The actual navigation data stops at "Alle Themen".

For pages, this is simpler (since they are not part of the navigation and therefore we only check if they are part of the navigation itself).

Reply:
So related to the question you had above: the dropdown case should only occur for pages, since the subject taxonomy terms shouldn't be a child of a dropdown.

Reply:
Thank you for the explaination, this makes much more sense now. So the takeaway is that the breadcrumbs are a combination of navigation *and* taxonomy - and the navigation nodes *could be* dropdowns without url.

Reply:
Exactly, totally obvious /s 😛 (yes, we love our legacy system)